jupyterlab_gcedetails/src/components/hardware_scaling_status.tsx

```diff
@@ -0,0 +1,173 @@
+import * as React from 'react';
+import { stylesheet } from 'typestyle';
+import LinearProgress from '@material-ui/core/LinearProgress';
+import { createStyles, withStyles, Theme } from '@material-ui/core/styles';
+import Button from '@material-ui/core/Button';
+import { ClientTransportService } from 'gcp_jupyterlab_shared';
+import { HardwareConfiguration, Details } from '../data';
+import { NotebooksService } from '../service/notebooks_service';
+import { AuthTokenRetrieval } from './auth_token_retrieval';
+
+const BorderLinearProgress = withStyles((theme: Theme) =>
+  createStyles({
+    root: {
+      width: 400,
+      height: 15,
+      borderRadius: 5,
+    },
+    colorPrimary: {
+      backgroundColor:
+        theme.palette.grey[theme.palette.type === 'light' ? 200 : 700],
+    },
+    bar: {
+      backgroundColor: '#1a90ff',
+      borderRadius: 5,
+    },
+  })
+)(LinearProgress);
+
+const STYLES = stylesheet({
+  flexContainer: {
+    width: 500,
+    height: 300,
+    display: 'flex',
+    flexDirection: 'column',
+    justifyContent: 'center',
+    alignItems: 'center',
+  },
+  heading: {
+    fontSize: '24px',
+  },
+  paragraph: {
+    height: 60,
+    width: 350,
+    textAlign: 'center',
+  },
+});
+
+enum Status {
+  'Authorizing' = 0,
+  'Stopping Instance' = 1,
+  'Reshaping Instance' = 2,
+  'Starting Instance' = 3,
+  'Complete' = 4,
+  'Error' = 5,
+}
+
+const statusInfo = [
+  'Getting authorization token to reshape machine. A dialog box should have popped up.',
+  'Shutting down instance for reshaping.',
+  'Reshaping instance to your configuration.',
+  'Restarting your instance. Your newly configured machine will be ready very shortly!',
+  'Operation complete. Enjoy your newly configured instance! You may now close this dialog.',
+  'An error has occured, please try again later. You may need to restart the instance manually.',
+];
+
+interface Props {
+  hardwareConfiguration: HardwareConfiguration;
+  onDialogClose: () => void;
+  details: Details;
+}
+
+interface State {
+  status: Status;
+}
+
+export class HardwareScalingStatus extends React.Component<Props, State> {
+  private notebookService: NotebooksService;
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      status: Status.Authorizing,
+    };
+    const clientTransportService = new ClientTransportService();
+    this.notebookService = new NotebooksService(clientTransportService);
```

Comment:
Instantiation in the constructor is generally an antipattern because it becomes almost impossible to unit test to mock out API calls.

Generally, you're going to want to create service classes that affect state and make API calls near the top-level of your application and then pass them down. Take a look at the GCP Scheduler where we create these classes at the time the extension is activated (https://github.com/GoogleCloudPlatform/jupyter-extensions/blob/e4e1df377fc5310d15dbe8e68a542afb452b9fcd/jupyterlab_gcpscheduler/src/index.ts#L83).

It may be useful to write some kind of utility function or class to obtain this instance if we find that we need to pass the service down through too many component layers. Happy to chat about this when we meet tomorrow.